include/crpropa/magneticField/turbulentField/PlaneWaveTurbulence.h

```diff
@@ -0,0 +1,110 @@
+#ifndef CRPROPA_PLANEWAVETURBULENCE_H
+#define CRPROPA_PLANEWAVETURBULENCE_H
+
+#include "crpropa/Grid.h"
+#include "crpropa/magneticField/turbulentField/TurbulentField.h"
+#include <vector>
+
+namespace crpropa {
+/**
+ * \addtogroup MagneticFields
+ * @{
+ */
+
+/**
+ @class PlaneWaveTurbulence
+ @brief Interpolation-free turbulent magnetic field based on the GJ99 and TD13
+papers
+
+ ## Overview
+ This class provides a turbulent magnetic field that is generated as described
+by [(Giacalone and Jokipii, 1999)][gj99] and [(Tautz and Dosch, 2013)][td13].
+Instead of using an inverse Fourier transform to generate the field on a grid --
+which then needs to be interpolated to obtain in-between values -- this method
+only generates the wave modes making up the turbulent magnetic field ahead of
+time. At run time, when the field's value at a particular position is required,
+these plane waves are then evaluated analytically at that position. This
+guarantees that the resulting field is completely free of divergence, reproduces
+the mean field strength accurately, and does not suffer from other
+interpolation-induced problems. The disadvantage is that the number of wave
+modes is drastically smaller when compared with initTurbulence, which might have
+physical ramifications on the particles propagating through the field.
+Additionally, the implementation is somewhat slower.
```

Comment:
Can we specify this "somewhat"? If not I would drop that sentence.

Reply:
I think that this might be a bit difficult say exactly. We do have some data on this in the paper, but in general this probably depends on a lot of factors – not just the number of wavemodes, but it will of probably also depend on which other modules are running and how much time they take, and we haven't done any testing on module configurations other than just PropagationCK.

Reply:
What about: "Furthermore, depending on the exact setting, the implementation is somewhat slower (see Schlegel et al. <link to the paper> for details)."

Reply:
That sounds good to me.

Reply:
Changed.